tests/fixtures/populate.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import subprocess
+import typing
+
+
+class Uploader(object):
+    def __init__(self, local_root: str):
+        self.local_root = local_root
+
+    def reset(self):
+        raise NotImplementedError()
+
+    def upload_file(self, local_path: str, remote_path: str, metadata_keys: typing.Dict[str, str]=dict()):
+        raise NotImplementedError()
+
+
+class S3Uploader(Uploader):
+    def __init__(self, local_root: str, bucket: str):
+        super(S3Uploader, self).__init__(local_root)
+        self.bucket = bucket
+
+    def reset(self):
+        args = [
+            "aws",
+            "s3",
+            "rm",
+            "s3://{}".format(self.bucket),
+            "--recursive"
+        ]
+        subprocess.check_call(args)
+
+    def upload_file(
+            self,
+            local_path: str,
+            remote_path: str,
+            metadata_keys: typing.Dict[str, str]=dict(),
+            tags: typing.Dict[str, str]=dict()):
+        args = [
+            "aws", "s3", "cp",
+            os.path.join(self.local_root, local_path),
+            "s3://{}/{}".format(self.bucket, remote_path),
+            "--metadata",
+            json.dumps(metadata_keys)
+        ]
+        subprocess.check_call(args)
+
+        tagset = dict(TagSet=[])
+        for tag_key, tag_value in tags.items():
+            tagset['TagSet'].append(
+                dict(
+                    Key=tag_key,
+                    Value=tag_value))
+        args = [
+            "aws",
+            "s3api",
+            "put-object-tagging",
+            "--bucket", self.bucket,
+            "--key", remote_path,
+            "--tagging", json.dumps(tagset)]
+        subprocess.check_call(args)
+
+
+class GSUploader(Uploader):
+    def __init__(self, local_root: str, bucket: str):
+        super(GSUploader, self).__init__(local_root)
+        self.bucket = bucket
+
+    def reset(self):
+        args = [
+            "gsutil",
+            "-m",
+            "rm",
+            "-r",
+            "gs://{}/**".format(self.bucket),
+        ]
+
+        # This is not a check_call because of https://github.com/GoogleCloudPlatform/gsutil/issues/417
+        subprocess.call(args)
+
+    def upload_file(self, local_path: str, remote_path: str, metadata_keys: typing.Dict[str, str]=dict()):
+        args = [
+            "gsutil",
+        ]
+        for metadata_key, metadata_value in metadata_keys.items():
+            args.extend([
+                "-h",
+                "x-goog-meta-{}:{}".format(metadata_key, metadata_value)
+            ])
+
+        args.extend([
+            "cp",
+            os.path.join(self.local_root, local_path),
+            "gs://{}/{}".format(self.bucket, remote_path),
+        ])
+        subprocess.check_call(args)
+
+
+def upload(uploader: Uploader):
```

Comment:
Please split the uploader code (which is generically useful) and the script that uploads a specific set of files (which isn't so much).